slinger/src/main/java/pl/allegro/android/slinger/Slinger.java

```diff
@@ -0,0 +1,63 @@
+package pl.allegro.android.slinger;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import java.util.Collections;
+import pl.allegro.android.slinger.enricher.DefaultIntentEnricher;
+import pl.allegro.android.slinger.enricher.IntentEnricher;
+import pl.allegro.android.slinger.resolver.IntentResolver;
+import pl.allegro.android.slinger.resolver.RedirectRule;
+
+public class Slinger {
+
+  private static IntentEnricher intentEnricher = new DefaultIntentEnricher();
+
+  /**
+   * Starts new {@link Intent} resolved by {@link IntentResolver}
+   * Starts passed intent and excludes <code>parentActivity</code>
+   *
+   * @param parentActivity that is used to be excluded from launch
+   * @param intent contains uri that is used to find a new {@link Intent}
+   */
+  public static void startActivity(Activity parentActivity, Intent intent) {
+    startActivity(parentActivity, intent, intentEnricher);
+  }
+
+  public static void startActivity(Activity parentActivity, Intent intent,
+      IntentEnricher intentEnricher) {
+    Uri uri = getOriginatingUriFromIntent(intent);
+
+    if (uri == null) {
+      throw new RuntimeException(
+          "You cannot run this Activity without specifying Uri inside Intent!");
+    }
+
+    excludeSlingerAndStartTargetActivity(parentActivity,
```

Comment:
any way to reformat this to improve readability?
extract method results to parameters?